examples/egress/egress-policy.yaml

```diff
@@ -0,0 +1,13 @@
+apiVersion: "projectcalico.org/v1"
+kind: Egresspolicy
```

Comment:
Related to my above comment, I'd like to make sure that this schema is flexible enough that we can extend it down the road with more features.

See this issue: https://github.com/projectcalico/calico/issues/458#issue-202650311

Specifically, this bit:

> I see two main options:

> 1. Make it look like the extensions.NetworkPolicy resource from the k8s API.
> 2. Make it look like the v1.Policy resource from the Calico API.

> I'm leaning towards option 2, since it will get us the full flexibility of the Calico API easier than converting to/from NetworkPolicy syntax, and probably needs less work to implement.

I'd be interested in your take (any anyone else who is interested in this feature!).  Does the schema defined [here](http://docs.projectcalico.org/v2.2/reference/calicoctl/resources/policy#sample-yaml) meet your needs, or is it important for this TPR to match the syntax of the Kubernetes API resource?

Let me mull on this and ask around, but deciding on the syntax of this TPR is the main question in my head at the moment.